File: homework01/caesar.py
def encrypt_caesar(plaintext):
    """
    Encrypt the message with Caesar cipher
    
    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    word = [i for i in plaintext]
    ciphertext = ''
    for i in range(len(word)):
        code_of_simbol = ord(word[i])
        if 65 <= code_of_simbol <= 87 or 97 <= code_of_simbol <= 119:
            word[i] = chr(code_of_simbol + 3)
        elif 88 <= code_of_simbol <= 90 or 120 <= code_of_simbol <= 122:
            word[i] = chr(code_of_simbol - 23)
        ciphertext += word[i]
    return ciphertext


def decrypt_caesar(ciphertext):
    """
    Decrypt the message with Caesar cipher
    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    word = [i for i in ciphertext]
    plaintext = ''
    for i in range(len(word)):
        code_of_simbol = ord(word[i])
        if 68 <= code_of_simbol <= 90 or 100 <= code_of_simbol <= 122:
            word[i] = chr(code_of_simbol - 3)
        elif 65 <= code_of_simbol <= 67 or 97 <= code_of_simbol <= 99:
            word[i] = chr(code_of_simbol + 23)
        plaintext += word[i]
    return plaintext

File: homework01/caesar.go
package main

import (
	"bytes"
)

func main() {
}

func encrypt_caesar(plaintext string) string {
	word := bytes.Runes([]byte(plaintext))
	ciphertext := ""
	for i := 0; i < len(word); i++ {
		if word[i] >= 65 && word[i] <= 87 || word[i] >= 97 && word[i] <= 119 {
			word[i] += 3
		} else if word[i] >= 88 && word[i] <= 90 || word[i] >= 120 && word[i] <= 122 {
			word[i] -= 23
		}
		ciphertext += string(word[i])
	}
	return ciphertext
}

func decrypt_caesar(ciphertext string) string {
	word := bytes.Runes([]byte(ciphertext))
	plaintext := ""
	for i := 0; i < len(word); i++ {
		if word[i] >= 68 && word[i] <= 90 || word[i] >= 100 && word[i] <= 122 {
			word[i] -= 3
		} else if word[i] >= 65 && word[i] <= 67 || word[i] >= 97 && word[i] <= 99 {
			word[i] += 23
		}
		plaintext += string(word[i])
	}
	return plaintext
}
